File: example/src/main/kotlin/com/example/_03_behavioral_patterns/_14_command/java/_02_after/GameEndCommand.java
package com.example._03_behavioral_patterns._14_command.java._02_after;

import com.example._03_behavioral_patterns._14_command.java._01_before.Game;

public class GameEndCommand implements Command {

    private final Game game;

    public GameEndCommand(Game game) {
        this.game = game;
    }

    @Override
    public void execute() {
        game.end();
    }

    @Override
    public void undo() {
        new GameStartCommand(this.game).execute();
    }
}

File: example/src/main/kotlin/com/example/_03_behavioral_patterns/_14_command/kt/_02_after/GameEndCommand.kt
package com.example._03_behavioral_patterns._14_command.kt._02_after

import com.example._03_behavioral_patterns._14_command.kt._01_before.Game

class GameEndCommand(
    private val game: Game
) : Command {

    override fun execute() {
        game.end()
    }

    override fun undo() {
        GameStartCommand(game).execute()
    }
}
